### PR TITLE
Add slog metadata for non-logger backends

### PR DESCRIPTION
**Pull request by k32**
Hello, @tolbrino 

I made a few improvements to my previous PR (1.3.0). The API became a bit more logger-ish.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -25,18 +25,22 @@ The idea for `hut` came out of
...
+  - `?slog(Level, Data :: map(), Metadata :: map())` structured log message with custom metadata
+  - `?set_process_metadata(Metadata :: map())` add custom metadata to the process
 4. Compile and pass the appropriate macro to the compiler to enable a certain backend as described next.
 5. Ensure that the application `hut` is started as part of your system.
 
 ## Supported Logging Backends
 
-- SASL `error_logger` as the default
+- SASL `error_logger` as the default (on OTP20 and older)
+- `logger ` as the default on OTP21+
```
Comment (tolbrino) [inline README.md]: One space too much at the end.

**Review by tolbrino — CHANGES_REQUESTED** [review]
Great change. Could you fix the issue for OTP 18? The macro `FUNCTION_NAME` has only been introduced in OTP 19.

Reference:

```

  OTP-13059    Application(s): compiler, stdlib

               The pre-processor can now expand the ?FUNCTION_NAME and
               ?FUNCTION_ARITY macros.
```

**Comment by k32** [comment]
Sure, will fix these issues later today.

**Comment by tolbrino** [comment]
> Sure, will fix these issues later today.

I removed OTP 18 from the list of supported versions, so please disregard my previous comment. 

If you could rebase on top of the latest master and remove that single space, the PR should be good.

File: Makefile [hunk 2]
```diff
@@ -1,6 +1,6 @@
 PROJECT = hut
 PROJECT_DESCRIPTION = helper library for making Erlang libraries logging framework agnostic
-PROJECT_VERSION = 1.3.0
+PROJECT_VERSION = 1.3.1
```
Comment (tolbrino) [inline Makefile]: Don't increment the version. This should only be done when releasing.

Comment (k32) [reply Makefile]: Rolled back

**Comment by k32** [comment]
Hello again. I applied the remarks to the best of my ability

**Review by tolbrino — APPROVED** [review]
LGTM, thanks.

**Comment by k32** [comment]
Thank you! Don't forget to make a hex release ;)

**Pull request merged**